### ISSUE TITLE
Back testing cross validation seems that is not working as documentation says

### ISSUE THREAD

**Issue by tinased95** [Nixtla/neuralforecast#520]
I reviewed the source code of the cross-validation module in core.py and noticed that each model in self.models only gets trained once, despite the documentation suggesting otherwise. I was expecting to see a loop for training and testing on different portions of the dataset, similar to the method used in [sklearn's backtesting method](https://scikit-learn.org/stable/modules/cross_validation.html#:~:text=%3E%3E%3E%20from%20sklearn.model_selection%20import%20TimeSeriesSplit,0%201%202%203%204%5D%20%5B5%5D). This inconsistency raises concerns about the accuracy of the cross-validation results.

Could you please clarify how the current implementation differs from the backtesting method in sklearn? I would appreciate any insights or explanations you can provide.

**Comment by FedericoGarza** [comment]
hey @tinased95! Thanks for using neuralforecast.

Sorry for the inconvenience caused by the documentation. Currently, neuralforecast performs cross-validation training only in a subset of the data and then produces rolled forecasts for the entire test set recursively without refitting (but taking into account the last observations of the particular window). The following image is more accurate:

<img width="629" alt="image" src="https://user-images.githubusercontent.com/10517170/232626468-af70a776-5273-4ba3-a9fd-05c58782c155.png">

We are working on including a `refit` argument to refit the model for each cross-validation window.

**Comment by cchallu** [comment]
Hi @tinased95. I posted an explanation on your issue #537.
